src/Nox.Types.EntityFramework/Configurations/NoxDatabaseConfigurator.cs

```diff
@@ -230,8 +233,47 @@ public abstract class NoxDatabaseConfigurator : INoxDatabaseConfigurator
                 databaseConfigurationForForeignKey.ConfigureEntityProperty(codeGeneratorState, builder, foreignEntityKeyDefinition, entity, false);
             }
         }
+        
+        protected virtual IList<IndexBuilder> ConfigureUniqueAttributeConstraints(IEntityBuilder builder, Entity entity)
+        {
+            var result = new List<IndexBuilder>();
+            if (entity.UniqueAttributeConstraints is not { Count: > 0 })
+            {
+                return result;
+            }
+            if (entity.Persistence!.IsAudited)
+            {
+                ConfigureConstraintsWithAuditProperties(builder, entity, result);
+            }
+            else
+            {
+                ConfigureConstraints(builder, entity, result);
+            }
+
+            return result;
+        }
+
+        private static void ConfigureConstraints(IEntityBuilder builder, Entity entity, List<IndexBuilder> result)
+        {
+            foreach (var uniqueConstraint in entity.UniqueAttributeConstraints!)
+            {
+                result.Add(builder.HasUniqueAttributeConstraint(uniqueConstraint.AttributeNames.ToArray(),
+                    uniqueConstraint.Name));
+            }
+        }
+
+        private static void ConfigureConstraintsWithAuditProperties(IEntityBuilder builder, Entity entity, List<IndexBuilder> result)
+        {
+            foreach (var uniqueConstraint in entity.UniqueAttributeConstraints!)
+            {
+                var auditProperties = new List<string>(uniqueConstraint.AttributeNames.Count + 1);
+                auditProperties.AddRange(uniqueConstraint.AttributeNames);
+                auditProperties.Add("DeletedAtUtc");
```

Comment:
use nameof(AuditableEntityBase.DeletedAtUtc)

Reply:
Since AuditableEntityBase is in Nox.Lib whic is not referenced from Nox.Types.EntityFramework, I have concerns about adding this reference. What do you think ?

Reply:
yup, you are correct, and that is related on Nox.EF is designed, thks